src/Pointer.ts

```diff
@@ -0,0 +1,50 @@
+import { fabric } from "fabric";
+import { makeLine } from "./Utils";
+import { Node } from "./Node";
+import Config = require("./Config");
+import { Variable } from "./Variable";
+
+
+export class Pointer {
```

Comment:
Love it, this is awesome. 

When we start doing more different types of Data Structures in the future, it might be a good time to use Typescript's template classes, which are awesome.